### ISSUE TITLE
amazon-books datasets

### ISSUE THREAD

**Issue by Iwillcome** [RUCAIBox/NCL#28]
when i try run code on amazon-books datasets base on amazon-books.yaml, the result is vastly different compared with the paper. The pre-procees dataset just contain the 4610 user node and 4138 item node, which is stange. while the origianl dataset is normal and unerror. i wish i can't the correct method to use amazon-boos. The detail descibe is fellow.
//////
General Hyper Parameters:
gpu_id = 1
use_gpu = True
seed = 2020
state = INFO
reproducibility = True
data_path = dataset/amazon-books
show_progress = True
save_dataset = False
save_dataloaders = False
benchmark_filename = None

Training Hyper Parameters:
checkpoint_dir = saved
epochs = 300
train_batch_size = 4096
learner = adam
learning_rate = 0.001
eval_step = 1
stopping_step = 10
clip_grad_norm = None
weight_decay = 0.0
loss_decimal_place = 4

Evaluation Hyper Parameters:
eval_args = {'split': {'RS': [0.8, 0.1, 0.1]}, 'group_by': 'user', 'order': 'RO', 'mode': 'full'}
metrics = ['Recall', 'NDCG']
topk = [10, 20, 50]
valid_metric = NDCG@10
valid_metric_bigger = True
eval_batch_size = 4096000
metric_decimal_place = 4

Dataset Hyper Parameters:
field_separator = 	
seq_separator =  
USER_ID_FIELD = user_id
ITEM_ID_FIELD = item_id
RATING_FIELD = rating
TIME_FIELD = timestamp
seq_len = None
LABEL_FIELD = label
threshold = None
NEG_PREFIX = neg_
load_col = {'inter': ['user_id', 'item_id', 'rating']}
unload_col = None
unused_col = None
additional_feat_suffix = None
rm_dup_inter = None
val_interval = {'rating': '[3,inf)'}
filter_inter_by_user_or_item = True
user_inter_num_interval = [15,inf)
item_inter_num_interval = [15,inf)
alias_of_user_id = None
alias_of_item_id = None
alias_of_entity_id = None
alias_of_relation_id = None
preload_weight = None
normalize_field = None
normalize_all = None
ITEM_LIST_LENGTH_FIELD = item_length
LIST_SUFFIX = _list
MAX_ITEM_LIST_LENGTH = 50
POSITION_FIELD = position_id
HEAD_ENTITY_ID_FIELD = head_id
TAIL_ENTITY_ID_FIELD = tail_id
RELATION_ID_FIELD = relation_id
ENTITY_ID_FIELD = entity_id

Other Hyper Parameters: 
neg_sampling = {'uniform': 1}
repeatable = False
MODEL_TYPE = ModelType.GENERAL
eval_setting = {'split': {'RS': [0.8, 0.1, 0.1]}, 'order': 'RO', 'group_by': 'user', 'mode': 'full'}
embedding_size = 64
n_layers = 3
reg_weight = 1e-06
ssl_temp = 0.05
ssl_reg = 1e-06
hyper_layers = 1
alpha = 0.8
proto_reg = 1e-07
num_clusters = 2000
m_step = 1
warm_up_step = 20
MODEL_INPUT_TYPE = InputType.PAIRWISE
eval_type = EvaluatorType.RANKING
device = cuda
train_neg_sample_args = {'strategy': 'by', 'by': 1, 'distribution': 'uniform'}
eval_neg_sample_args = {'strategy': 'full', 'distribution': 'uniform'}

28 Nov 20:25    INFO  amazon-books
The number of users: 4610
Average actions of users: 34.132566717292256
The number of items: 4138
Average actions of items: 38.026831036983324
The number of inters: 157317
The sparsity of the dataset: 99.17532231295783%
Remain Fields: ['user_id', 'item_id', 'rating']
////

**Comment by linzihan-backforward** [comment]
Hi, we have verified the amazon-books.yaml and run the code locally. The dataset information is consistent with our paper. Please make sure that the version of RecBole is 1.0.1 and the raw dataset is correct. If you still have the problem, please provide more details such as the version, configuration and dataset.

**Comment by Iwillcome** [comment]
Dear zhao：
   Thanks for you response, Too busy work to forget many importance thing. The version of RecBole is 1.0.0. After updata it to 1.0.1, and i downing again amazon-book dataset by aotu-down tool in recBole,.At this time, some different error occur, the number of node over 50000 when other parameters keep. 
07 Dec 15:28    INFO  
General Hyper Parameters:
gpu_id = 0
use_gpu = True
seed = 2020
state = INFO
reproducibility = True
data_path = dataset/amazon-books
checkpoint_dir = saved
show_progress = True
save_dataset = False
dataset_save_path = None
save_dataloaders = False
dataloaders_save_path = None
log_wandb = False

Training Hyper Parameters:
epochs = 300
train_batch_size = 1024
learner = adam
learning_rate = 0.001
neg_sampling = {'uniform': 1}
eval_step = 1
stopping_step = 10
clip_grad_norm = None
weight_decay = 0.0
loss_decimal_place = 4

Evaluation Hyper Parameters:
eval_args = {'split': {'RS': [0.8, 0.1, 0.1]}, 'group_by': 'user', 'order': 'RO', 'mode': 'full'}
repeatable = False
metrics = ['Recall', 'NDCG']
topk = [10, 20, 50]
valid_metric = NDCG@10
valid_metric_bigger = True
eval_batch_size = 4096000
metric_decimal_place = 4

Dataset Hyper Parameters:
field_separator = 	
seq_separator =  
USER_ID_FIELD = user_id
ITEM_ID_FIELD = item_id
RATING_FIELD = rating
TIME_FIELD = timestamp
seq_len = None
LABEL_FIELD = label
threshold = None
NEG_PREFIX = neg_
load_col = {'inter': ['user_id', 'item_id', 'rating']}
unload_col = None
unused_col = None
additional_feat_suffix = None
rm_dup_inter = None
val_interval = {'rating': '[3,inf)'}
filter_inter_by_user_or_item = True
user_inter_num_interval = [20,inf)
item_inter_num_interval = [20,inf)
alias_of_user_id = None
alias_of_item_id = None
alias_of_entity_id = None
alias_of_relation_id = None
preload_weight = None
normalize_field = None
normalize_all = None
ITEM_LIST_LENGTH_FIELD = item_length
LIST_SUFFIX = _list
MAX_ITEM_LIST_LENGTH = 50
POSITION_FIELD = position_id
HEAD_ENTITY_ID_FIELD = head_id
TAIL_ENTITY_ID_FIELD = tail_id
RELATION_ID_FIELD = relation_id
ENTITY_ID_FIELD = entity_id
benchmark_filename = None

Other Hyper Parameters: 
wandb_project = recbole
require_pow = False
MODEL_TYPE = ModelType.GENERAL
eval_setting = {'split': {'RS': [0.8, 0.1, 0.1]}, 'order': 'RO', 'group_by': 'user', 'mode': 'full'}
embedding_size = 64
n_layers = 3
reg_weight = 1e-06
ssl_temp = 0.05
ssl_reg = 1e-06
hyper_layers = 1
alpha = 0.8
proto_reg = 1e-07
num_clusters = 2000
m_step = 1
warm_up_step = 20
MODEL_INPUT_TYPE = InputType.PAIRWISE
eval_type = EvaluatorType.RANKING
device = cuda
train_neg_sample_args = {'strategy': 'by', 'by': 1, 'distribution': 'uniform', 'dynamic': 'none'}
eval_neg_sample_args = {'strategy': 'full', 'distribution': 'uniform'}


07 Dec 15:31    INFO  amazon-books
The number of users: 30211
Average actions of users: 54.50625620655412
The number of items: 33177
Average actions of items: 49.63328912466844
The number of inters: 1646634
The sparsity of the dataset: 99.83571615269378%
Remain Fields: ['user_id', 'item_id', 'rating']
07 Dec 15:31    INFO  [Training]: train_batch_size = [1024] negative sampling: [{'uniform': 1}]
07 Dec 15:31    INFO  [Evaluation]: eval_batch_size = [4096000] eval_args: [{'split': {'RS': [0.8, 0.1, 0.1]}, 'group_by': 'user', 'order': 'RO', 'mode': 'full'}]

meanwhile, The result at the time is different two days in the same sever mechine. The number of user node just 30000.

**Comment by linzihan-backforward** [comment]
> We filter out users and itemswith fewer than 15 interactions by the command：
`user_inter_num_interval: "[15,inf)" 
item_inter_num_interval: "[15,inf)"`
In this way, the dataset information would be correct.

**Comment by Iwillcome** [comment]
Dear Auther:
     Thanks for you response. We reconfigured the file and found that there was no update. The result of the redownloaded dataset is the same as that of the paper after processing. My problem has been solved.
                                                                                                                                                     Best Wish!
